[1.12.2] Nether is being disabled incorrectly

<!-- ISSUE_TEMPLATE_3 -> IMPORTANT: DO NOT DELETE THIS LINE.-->

<!-- Thank you for reporting ! Please note that issues can take a lot of time to be fixed and there is no eta.-->

<!-- If you don't know where to upload your logs and crash reports, you can use these websites : -->
<!-- https://gist.github.com (recommended) -->
<!-- https://mclo.gs -->
<!-- https://haste.mohistmc.com -->
<!-- https://pastebin.com -->

<!-- TO FILL THIS TEMPLATE, YOU NEED TO REPLACE THE {} BY WHAT YOU WANT -->

**Minecraft Version :** 1.12.2

**Mohist Version :** 295

**Operating System :** Linux Ubuntu Server 21.10

**Logs :** [latest.log](https://github.com/MohistMC/Mohist/files/7809105/latest.log)

**Mod list :** https://haste.mohistmc.com/nadonudeci (no mods)

**Plugin list :** https://haste.mohistmc.com/akorowosor (no plugins)

**Description of issue :** If you set the "allow-nether" parameter to false in server.properties and try to log in to Nether through the portal, then the console will be spammed with this error
```
[19:51:49] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_311]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_311]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:840) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:461) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:778) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:624) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_311]
Caused by: java.lang.NullPointerException
	at net.minecraft.network.NetHandlerPlayServer.func_147347_a(NetHandlerPlayServer.java:770) ~[pa.class:?]
	at net.minecraft.network.play.client.CPacketPlayer.func_148833_a(CPacketPlayer.java:33) ~[lk.class:?]
	at net.minecraft.network.play.client.CPacketPlayer$Position.func_148833_a(CPacketPlayer.java:76) ~[lk$a.class:?]
	at net.minecraft.network.PacketThreadUtil.lambda$checkThreadAndEnqueue$0(PacketThreadUtil.java:16) ~[hv.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_311]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_311]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 5 more
```
There is no such problem on mohist version 162, but I don't know exactly from which version this problem appeared

--- a/patches/minecraft/net/minecraft/server/management/PlayerList.java.patch
+++ b/patches/minecraft/net/minecraft/server/management/PlayerList.java.patch
@@ -906,7 +906,7 @@
          return entityplayermp;
      }
  
-@@ -556,15 +1106,80 @@
+@@ -556,15 +1106,81 @@
  
      public void func_187242_a(EntityPlayerMP p_187242_1_, int p_187242_2_)
      {
@@ -931,9 +931,7 @@
 +        PlayerTeleportEvent.TeleportCause cause = (teleportCause == null ? PlayerTeleportEvent.TeleportCause.MOD : teleportCause);
          int i = p_187242_1_.field_71093_bK;
          WorldServer worldserver = this.field_72400_f.func_71218_a(p_187242_1_.field_71093_bK);
-         p_187242_1_.field_71093_bK = p_187242_2_;
-         WorldServer worldserver1 = this.field_72400_f.func_71218_a(p_187242_1_.field_71093_bK);
--        p_187242_1_.field_71135_a.func_147359_a(new SPacketRespawn(p_187242_1_.field_71093_bK, p_187242_1_.field_70170_p.func_175659_aa(), p_187242_1_.field_70170_p.func_72912_H().func_76067_t(), p_187242_1_.field_71134_c.func_73081_b()));
++        WorldServer worldserver1 = this.field_72400_f.func_71218_a(p_187242_2_);
 +
 +        // CraftBukkit start - Replaced the standard handling of portals with a more customised method.
 +        Location enter = p_187242_1_.getBukkitEntity().getLocation();
@@ -968,6 +966,10 @@
 +            return;
 +        }
 +
+         p_187242_1_.field_71093_bK = p_187242_2_;
+-        WorldServer worldserver1 = this.field_72400_f.func_71218_a(p_187242_1_.field_71093_bK);
+-        p_187242_1_.field_71135_a.func_147359_a(new SPacketRespawn(p_187242_1_.field_71093_bK, p_187242_1_.field_70170_p.func_175659_aa(), p_187242_1_.field_70170_p.func_72912_H().func_76067_t(), p_187242_1_.field_71134_c.func_73081_b()));
++
 +        Vector velocity = p_187242_1_.getBukkitEntity().getVelocity();
 +        worldserver1.func_85176_s().adjustExit(p_187242_1_, exit, velocity);
 +
@@ -989,7 +991,7 @@
          this.func_72375_a(p_187242_1_, worldserver);
          p_187242_1_.field_71135_a.func_147364_a(p_187242_1_.field_70165_t, p_187242_1_.field_70163_u, p_187242_1_.field_70161_v, p_187242_1_.field_70177_z, p_187242_1_.field_70125_A);
          p_187242_1_.field_71134_c.func_73080_a(worldserver1);
-@@ -576,17 +1191,39 @@
+@@ -576,17 +1192,39 @@
          {
              p_187242_1_.field_71135_a.func_147359_a(new SPacketEntityEffect(p_187242_1_.func_145782_y(), potioneffect));
          }
@@ -1032,7 +1034,7 @@
          {
              d0 = MathHelper.func_151237_a(d0 / 8.0D, p_82448_4_.func_175723_af().func_177726_b() + 16.0D, p_82448_4_.func_175723_af().func_177728_d() - 16.0D);
              d1 = MathHelper.func_151237_a(d1 / 8.0D, p_82448_4_.func_175723_af().func_177736_c() + 16.0D, p_82448_4_.func_175723_af().func_177733_e() - 16.0D);
-@@ -597,7 +1234,7 @@
+@@ -597,7 +1235,7 @@
                  p_82448_3_.func_72866_a(p_82448_1_, false);
              }
          }
@@ -1041,7 +1043,7 @@
          {
              d0 = MathHelper.func_151237_a(d0 * 8.0D, p_82448_4_.func_175723_af().func_177726_b() + 16.0D, p_82448_4_.func_175723_af().func_177728_d() - 16.0D);
              d1 = MathHelper.func_151237_a(d1 * 8.0D, p_82448_4_.func_175723_af().func_177736_c() + 16.0D, p_82448_4_.func_175723_af().func_177733_e() - 16.0D);
-@@ -608,7 +1245,7 @@
+@@ -608,7 +1246,7 @@
                  p_82448_3_.func_72866_a(p_82448_1_, false);
              }
          }
@@ -1050,7 +1052,7 @@
          {
              BlockPos blockpos;
  
-@@ -634,7 +1271,7 @@
+@@ -634,7 +1272,7 @@
  
          p_82448_3_.field_72984_F.func_76319_b();
  
@@ -1059,7 +1061,7 @@
          {
              p_82448_3_.field_72984_F.func_76320_a("placing");
              d0 = (double)MathHelper.func_76125_a((int)d0, -29999872, 29999872);
-@@ -643,7 +1280,8 @@
+@@ -643,7 +1281,8 @@
              if (p_82448_1_.func_70089_S())
              {
                  p_82448_1_.func_70012_b(d0, p_82448_1_.field_70163_u, d1, p_82448_1_.field_70177_z, p_82448_1_.field_70125_A);
@@ -1069,7 +1071,7 @@
                  p_82448_4_.func_72838_d(p_82448_1_);
                  p_82448_4_.func_72866_a(p_82448_1_, false);
              }
-@@ -654,11 +1292,106 @@
+@@ -654,11 +1293,106 @@
          p_82448_1_.func_70029_a(p_82448_4_);
      }
  
@@ -1177,7 +1179,7 @@
              this.field_72408_o = 0;
          }
      }
-@@ -671,6 +1404,24 @@
+@@ -671,6 +1405,24 @@
          }
      }
  
@@ -1202,7 +1204,7 @@
      public void func_148537_a(Packet<?> p_148537_1_, int p_148537_2_)
      {
          for (int i = 0; i < this.field_72404_b.size(); ++i)
-@@ -786,12 +1537,20 @@
+@@ -786,12 +1538,20 @@
          int i = this.field_72400_f.func_110455_j();
          this.field_72414_i.func_152687_a(new UserListOpsEntry(p_152605_1_, this.field_72400_f.func_110455_j(), this.field_72414_i.func_183026_b(p_152605_1_)));
          this.func_187245_a(this.func_177451_a(p_152605_1_.getId()), i);
@@ -1223,7 +1225,7 @@
      }
  
      private void func_187245_a(EntityPlayerMP p_187245_1_, int p_187245_2_)
-@@ -847,6 +1606,12 @@
+@@ -847,6 +1607,12 @@
          {
              EntityPlayerMP entityplayermp = this.field_72404_b.get(i);
  
@@ -1236,7 +1238,7 @@
              if (entityplayermp != p_148543_1_ && entityplayermp.field_71093_bK == p_148543_10_)
              {
                  double d0 = p_148543_2_ - entityplayermp.field_70165_t;
-@@ -863,12 +1628,26 @@
+@@ -863,12 +1629,26 @@
  
      public void func_72389_g()
      {
@@ -1267,7 +1269,7 @@
      public void func_152601_d(GameProfile p_152601_1_)
      {
          this.field_72411_j.func_152687_a(new UserListWhitelistEntry(p_152601_1_));
-@@ -913,16 +1692,21 @@
+@@ -913,16 +1693,21 @@
  
          if (p_72354_2_.func_72896_J())
          {
@@ -1293,7 +1295,7 @@
          p_72385_1_.field_71135_a.func_147359_a(new SPacketHeldItemChange(p_72385_1_.field_71071_by.field_70461_c));
      }
  
-@@ -1004,17 +1788,26 @@
+@@ -1004,17 +1789,26 @@
  
      public void func_72392_r()
      {
@@ -1324,7 +1326,7 @@
      }
  
      public void func_148539_a(ITextComponent p_148539_1_)
-@@ -1022,6 +1815,10 @@
+@@ -1022,6 +1816,10 @@
          this.func_148544_a(p_148539_1_, true);
      }
  
@@ -1335,7 +1337,7 @@
      public StatisticsManagerServer func_152602_a(EntityPlayer p_152602_1_)
      {
          UUID uuid = p_152602_1_.func_110124_au();
-@@ -1073,7 +1870,7 @@
+@@ -1073,7 +1871,7 @@
  
          if (this.field_72400_f.field_71305_c != null)
          {

